src/flowr/internal-session.ts

```diff
@@ -84,7 +69,19 @@ export class FlowrInternalSession {
 			return a.location.start.line - b.location.start.line || a.location.start.column - b.location.start.column
 		})
 
-		const diagnostics: vscode.Diagnostic[] = []
+		this.outputChannel.appendLine('slice: ' + JSON.stringify([...result.slice.result]))
+		this.diagnostics.set(uri, FlowrInternalSession.createDiagnostics(document, range, pos, sliceElements))
+		return result.reconstruct.code
+	}
+
+	public static getPositionAt(position: vscode.Position, document: vscode.TextDocument): vscode.Range | undefined {
+		const re = /([a-zA-Z0-9._:])+/
```

Comment:
again, a really.... intersting way to get "the beginning" of a token - which does, for example, not work if the token is escaped. so we have two options

1) make this much more sophisticated and invest like 20 hours
2) handle the issue on the flowr side allowing a slicing point to be specified "looseley" by position (i.e. add another slicing point format which does not require the first position

i am for 2

Reply:
why are you criticizing your own code as if it was my fault that it's bad <3

Reply:
https://github.com/Code-Inspect/flowr/issues/704